### PR TITLE
Creating and delivering promise for kickoff pipeline

### PR DESCRIPTION
**Pull request by ericakresch**

### CODE REVIEW

**Review by kocubinski — APPROVED** [review]

**Review by Scriptor — APPROVED** [review]
Looks good! Having this just return the channels before was purely done for debugging/performance testing reasons, so this is a step in the right direction.

**Comment by mgaare** [comment]
Gonna probably need to bump the minor version on this since the return signature of one of the main API functions is different now.

**Comment by mgaare** [comment]
> Looks good! Having this just return the channels before was purely done for debugging/performance testing reasons, so this is a step in the right direction.

Actually since it's changing up anyway, maybe have it return a map instead of a tuple, so maybe like:
`{:chans chans
  :completion complete}`

File: src/org/purefn/sentenza/api.clj [hunk 1]
```diff
@@ -195,14 +195,16 @@
...
    (let [state  (proto/init pipeline args)
          source (channeled (proto/source pipeline state) n)
-         chs    (proto/run pipeline state source)]
+         chs    (proto/run pipeline state source)
+         completed (promise)]
      (go-loop []
        (if (nil? (<! (last chs)))
          (do
            (close! (last chs))
+           (deliver completed :done)
```
Comment (mgaare) [inline src/org/purefn/sentenza/api.clj]: Hm, it might be better to wait on delivering the promise until after the cleanup call.

**Comment by ericakresch** [comment]
@mgaare makes sense. Made those updates.

**Pull request merged**